src/darkleaf/di/core.clj

```diff
@@ -55,10 +55,11 @@
 
 (declare find-or-build)
 
-(defn- missing-dependency! [key]
+(defn- missing-dependency! [{:as ctx :keys [*current-key]} key]
   (throw (ex-info (str "Missing dependency " key)
-                  {:type ::missing-dependency
-                   :key  key})))
+                  {:type   ::missing-dependency
+                   :key    key
+                   :parent *current-key})))
```

Comment:
may be you could just use `:under-construction` ?

Reply:
@darkleaf :under-construction is a set, you can't get last added element from it